web-api/openapi/front_office.yaml

```diff
@@ -129,6 +129,16 @@ paths:
             minimum: 1
             maximum: 100
             default: 10
+        - name: min_similarity
+          in: query
+          description: Minimum similarity a document has to have to be included.
+          required: false
+          schema:
+            type: number
+            format: float
+            minimum: 0
```

Comment:
:wrench: we should set the minimum to `-1`. if someone provides this parameter but still wants to include all documents, then that wouldn't be possible with minimum `0`. it's likely not a common use case, but there is no reason to restrict this artificially.

Reply:
AFIK scores of ES KNN search are between 0 and 1 so 0 should always include all values :thinking:

Reply:
ah i see, then there is a misunderstanding, the elastic score is indeed in `[0, 1]`. but since you named the parameter `min_similarity`, i automatically assumed that you are refererring to cosine similarity (i could have checked better the code using the param). to avoid such confusion also for others, i guess it's best to choose a different name for the param to better reflect its purpose. `min_score` would be an option, maybe also sth else.

Reply:
The problem with `min_score` is that it's not the min score of `PersonalizedDocument.score` but a threshold applied to the similarity search only.  And as far as I can tell the score of ES is a similarity of some form, probably a `[-1;1]` similarity normalized to `[0;1]`. 

Also the mathematical properties of cosin similarity and similar should not be aspects our customers should worry about. The customer just knows there is some similarity search followed by some personalization and the threshold is on the similarity search.

Reply:
Even for the scoring of the reranker we do there are reasons why we might want to transform it from `[-1;1]` to `[0;1]` depending on how we might want to handle certain things like negative reactions and differentiating between very dissimilar to a CoI (doesn't mean the user dislikes it, could be returned if we have nothing better) and disliked by a user (should mostly not be returned).

Reply:
normalization of scores is indeed a good point. so i guess we should leave this for now and handle it in a follow up 👌